miasm/core/cpu.py

```diff
@@ -1214,7 +1219,10 @@ def dis(cls, bs_o, mode_o = None, offset=0):
             total_l = 0
             for i, f in enumerate(c.fields_order):
                 if f.flen is not None:
-                    l = f.flen(mode, fname_values)
+                    try:
+                        l = f.flen(mode, fname_values, bs, offset_b)
```

Comment:
Same here: flen may raise IOError, so you have to handle it like an IOError in getbits (I mean closing the atomic mode...)

Reply:
I fixed it in the latest commit too.